pulp_ansible/app/serializers.py

```diff
@@ -648,11 +811,51 @@ class CollectionVersionSerializer(ContentChecksumSerializer, CollectionVersionUp
 
     creating = True
 
+    def create(self, validated_data):
+        artifact = validated_data.get("artifact")
+        repository = validated_data.get("repository")
+        sigstore_verifying_services = repository.sigstore_verifying_service.all()
+
+        cv_instance = super().create(validated_data)
+
+        artifact_name = artifact.file.name
+        artifact_file = storage.open(artifact_name)
```

Comment:
I really don't like that we create the CV instance from the artifact, doing a bunch of processing on the tarball, only to then again do more processing afterward to figure out of a signature is inside. It would be great if we could do this processing here: https://github.com/pulp/pulp_ansible/blob/main/pulp_ansible/app/tasks/upload.py#L17 and then maybe the final Signature creation in `finish_collection_upload` (although would need to find some way to also get the signature inside the repository as well...)

Reply:
Yes, this indeed requires extra processing which I don't like either. However I couldn't figure out another place where to find a reference to the upload repository to upload the signature there as well. Do you think we could leave the signature creation process as it is now or do you see any alternative for the collection to be processed just once and still have eventual signatures uploaded to the right repo?

Reply:
I think you can move this process code into `process_collection_artifact` and have the found signature bundle returned as a part of the collection info. Then on the create function (https://github.com/pulp/pulp_ansible/blob/main/pulp_ansible/app/serializers.py#L524-L529) you pop off the signature information and if it exists then you create the signatures, verifying with the repository's signing verifiers. This way we don't double process the artifact.